icicle/tests/test_hash_api.cpp

```diff
@@ -165,32 +165,1090 @@ TEST_F(HashApiTest, MerkleTree)
   ASSERT_TRUE(verification_valid);
 }
 
-#ifdef POSEIDON
+// #ifdef POSEIDON
 
 #include "icicle/fields/field_config.h"
 using namespace field_config;
 
 #include "icicle/hash/poseidon.h"
 
-TEST_F(HashApiTest, poseidon12)
+TEST_F(HashApiTest, poseidon3)
 {
-  const uint64_t arity = 12; // Number of input elements
+  const uint64_t arity = 3; // Number of input elements
 
   // Create unique pointers for input and output arrays
   auto input = std::make_unique<scalar_t[]>(arity);
   scalar_t output = scalar_t::from(0);
   // Randomize the input array
-  scalar_t::rand_host_many(input.get(), arity);
+  input[0] = scalar_t::from(1);
+  input[1] = scalar_t::from(1);
+  input[2] = scalar_t::from(1);
 
-  // init poseidon constants on current device
-  ICICLE_CHECK(Poseidon::init_default_constants<scalar_t>());
+  // Create Poseidon hash object
+  auto poseidon = Poseidon::create<scalar_t>(arity);
+
+  // Run single hash operation
+  auto config = default_hash_config();
+  ICICLE_CHECK(poseidon.hash(input.get(), arity, config, &output));
+  
+  std::cout << "output: " << output << std::endl;
+  // TODO: Verify output (e.g., check CPU against CUDA)
+}
+
+TEST_F(HashApiTest, poseidon3_custom_constants)
+{
+  const uint64_t arity = 3; // Number of input elements
+
+  // Create unique pointers for input and output arrays
+  auto input = std::make_unique<scalar_t[]>(arity);
+  scalar_t output = scalar_t::from(0);
+  // Randomize the input array
+  input[0] = scalar_t::from(1);
+  input[1] = scalar_t::from(1);
+  input[2] = scalar_t::from(1);
+
+  PoseidonConstantsInitOptions<scalar_t> options = {};
+
+  unsigned char poseidon_constants_2[] = {
```

Comment:
can you maybe randomize it? if we compare CPU to CUDA it should work with any constants right? it should be a more robust test too.